### PR TITLE
Fixed #59395 - Emmet Syntax Profiles tag_nl produces no extra space

### LINKED ISSUE #59395: Emmet Syntax Profiles tag_nl adds extra white space
- VSCode Version: 1.27.2
- OS Version: Ubuntu 18.04

Steps to Reproduce:

1. Add this to my User Settings:

```
   "emmet.syntaxProfiles": {
        "html": {
            "tag_nl": false            
        }
    },
```
2. Type some text: foo

 3. Select emmet: Wrap with Abbreviation and select a tag such as h1

Expected: **&lt;h1&gt;foo&lt;/h1&gt;**
Actual: 

```
<h1> 	foo </h1>
```
Note the extra white space.

Does this issue occur when all extensions are disabled?: Yes/No 

Not sure how to answer this as I believe emmet is an extension.

### PR DESCRIPTION
**Pull request by irrationalRock**
This PR fixes https://github.com/Microsoft/vscode/issues/59395.

With:
```
"emmet.syntaxProfiles": {
        "html": {
            "tag_nl": false            
        }
    },
```

It no longer allows extra space.

Before:
vscode-59395-before

After:
vscode-59395

Thanks for considering this request.

### CODE REVIEW

File: extensions/emmet/src/abbreviationActions.ts [hunk 1]
```diff
@@ -613,8 +613,10 @@ function expandAbbr(input: ExpandAbbreviationInput): string {
 					wrappingNode = wrappingNode.children[wrappingNode.children.length - 1];
 				}
 
+				const emmetConfig = getEmmetConfiguration(input.syntax);
+
 				// If wrapping with a block element, insert newline in the text to wrap.
-				if (wrappingNode && inlineElements.indexOf(wrappingNode.name) === -1) {
+				if (wrappingNode && inlineElements.indexOf(wrappingNode.name) === -1 && emmetConfig.syntaxProfiles['html']['tag_nl']) {
```
Comment (ramya-rao-a) [inline extensions/emmet/src/abbreviationActions.ts]: We can probably get the value for `tag_nl` from `expandOptions.profile` object

**Review by ramya-rao-a — CHANGES_REQUESTED** [review]
The tests seem to be failing. Can you take a look at that?

To run the tests locally, run `scripts\test-integration.bat` on Windows or `./scripts/test-integration.sh` on Linux/Mac

It would also do us good to add a test case to cover this scenario.
`wrapWithAbbreviation.test.ts` would be the file to update.

**Comment by irrationalRock** [comment]
I updated my code to use `expandOptions.profile`. I'm wondering if you can give me an example of how to change the config in tests. I am having some problems trying to change `tag_nl` to `false`.

**Review by ramya-rao-a — CHANGES_REQUESTED** [review]
My apologies, I should have cleared some things previously.

The new emmet modules that we use dont use the same `syntaxProfile` settings as documented in https://docs.emmet.io/customization/syntax-profiles/

The new emmet modules follow the profile as defined in https://github.com/emmetio/output-profile/blob/1a7571e78da9d7bf056289b3396dbb0bcc45c435/types.d.ts#L1

To support backward compatibility in VS Code, we support both in the settings. I convert the old profile to a new one using a [helper](https://github.com/Microsoft/vscode-emmet-helper/blob/7ec2e95ea7caea3d938e70cc194a9bb4eebd05e3/src/emmetHelper.ts#L689)

So now coming to the current issue. You can just add `expandOptions['format'] === true` to the `if` condition and that should be enough.

**Comment by ramya-rao-a** [comment]
https://github.com/Microsoft/vscode/blob/release/1.28/extensions/emmet/src/test/abbreviationAction.test.ts#L441 is an example where I changed the config for the `emmet.excludeLanguages` setting. You can use that as an example.

**Comment by irrationalRock** [comment]
Thanks for the help. I had to use `expandOptions['profile'].format` instead of `expandOptions['format']` because it didn't seem to produce any properties when `tag_nl` was set to `true`.
I also added tests for this feature.

**Comment by irrationalRock** [comment]
It seems like my commits fail when running the hygiene tests. Is there any way for me to run these tests on my local machine? Running `./scripts/test-integration.sh` passes all of the tests.

**Comment by ramya-rao-a** [comment]
The hygiene failure was due to formatting errors. I have pushed a change to fix that.

**Review by ramya-rao-a — APPROVED** [review]

**Comment by contentfree** [comment]
Wrap still produces incorrect results if `tag_nl_leaf` is false, I believe. (`tag_nl_leaf` doesn't seem to be respected at all, really, nor does the `'decide'` value for `tag_nl`)

**Comment by ramya-rao-a** [comment]
@contentfree I have logged an upstream issue for that. See https://github.com/emmetio/output-profile/issues/1

**Pull request merged**